### PR TITLE
Makefile: Add `clean_builtsrc`, `clean_builtassets`

### PR DESCRIPTION
**Pull request by Dragorn421**
# TLDR

This PR adds `clean_builtsrc` (to rebuild `src/`),
and `clean_builtassets` (to rebuild `assets/`)
`make clean && make` : 3min40s
`make clean_builtsrc && make` : 1min40s
`make clean_builtassets && make` : 2min10s

# Motivation

`make clean` is often used when building (`make`) fails for some obscure reason, or the ROM doesn't build OK for equally obscure reasons

Building from clean (`make` after `make clean`) takes a long time, 3min40s for me without `-j`.
The "clean" operation can however be separated into two parts: cleaning built assets (in `build/assets`) and built source (in `build/src`)
It is very rarely useful or required to rebuild all of both of these, and typically rebuilding all of either is enough

This PR adds `clean_builtsrc` and `clean_builtassets` alongside `clean`.
Where `clean` removes the whole `build` folder of intermediate files, `clean_builtsrc` focuses on `build/src` and `clean_builtassets` on `build/assets` (they both also remove other things `clean` removes, see the `Makefile`)

Building from `clean_builtsrc` (i.e. re-building `src/`) is 1min40s for me, half the time to build from clean (still `make` without `-j`)
And building from `clean_builtassets` (i.e. re-building `assets/`) is 2min10s for me

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -251,6 +251,14 @@ endif
 clean:
 	$(RM) -r $(ROM) $(ELF) build
 
+clean_builtsrc:
+	$(RM) -r $(ROM) $(ELF) build/src build/data
+	find build -maxdepth 1 -type f -exec rm {} \;
```
[the remark this answers is not in the document]
Comment (Dragorn421) [reply Makefile:256]: It's removing files directly inside `build/`, i.e.
`dmadata_table_spec.h  ldscript.txt  spec   undefined_syms.txt  z64.map` (in `build/`)

I could also put that list of files into the makefile tbh but it felt less random to remove all files

[the remark this answers is not in the document]
Comment (Dragorn421) [reply Makefile:256]: (fwiw this is not advanced find usage at all, assuming this may get in and if we keep it I'll add a comment on it though)

File: Makefile [hunk 2]
```diff
@@ -251,6 +251,15 @@ endif
 clean:
 	$(RM) -r $(ROM) $(ELF) build
 
+clean_builtsrc:
+	$(RM) -r $(ROM) $(ELF) build/src build/data
+# Delete files directly inside build/ (but not in subfolders)
+	find build -maxdepth 1 -type f -exec $(RM) {} \;
+
+clean_builtassets:
```
Comment (hensldm) [inline Makefile:259]:
```suggestion
builtassetsclean:
```
Wonder if this would be better to mirror `assetclean`? (same idea with `clean_builtsrc`)
Or perhaps `builtassets_clean` and `assets_clean`?

**Review by hensldm — COMMENTED** [review]
I'm fine with this implementation, but thoughts on instead of new targets, use clean with an env variable (i.e. `clean TARGET=builtsrc`). I'll admit does seem a little more clunky.

**Comment by vitaninten** [comment]
I beat the shadow temple boss when I did this but should I get the biggoron and how?

**Comment by Dragorn421** [comment]
I don't think this is going anywhere, so, closing
